### PR TITLE
Simplify logging configuration.

### PR DESCRIPTION
**Pull request by mbarnes**
This simplifies logging configuration, both in the file format and internally.

For starters, we return to the `logging.config` schema in the JSON configuration file, under the `"logging"` key.  But this should only be necessary for changing specific loggers or log message handling.

For convenience, `DEBUG` level log messages can be enabled for all loggers by setting a `"debug"` option to `true` in the configuration file.

Internally, rewrite `setup_logging()` to install a default handler for the root logger only, instead of for each and every non-root logger.  Non-root loggers will most likely propagate messages to the root logger, unless the configuration overrides this.

This also eliminates the maintenance cost of having to specify all logger names when calling `setup_logging()`.

### CODE REVIEW

**Comment by ashcrow** [comment]
Looks like unittests are failing.

File: src/commissaire/util/logging.py [hunk 1]
```diff
@@ -15,62 +15,40 @@
...
+        }
+
+    handlers = config.setdefault('handlers', {})
+    if 'default' not in handlers:
+        handlers['default'] = {
+            'class': 'logging.StreamHandler',
+            'formatter': 'default'
+        }
+
+    root_logger = config.setdefault('root', {})
```
Comment (ashcrow) [inline src/commissaire/util/logging.py:49]: Is ``root`` expected to be used by default, or is it for fallback?

Comment (mbarnes) [reply src/commissaire/util/logging.py:49]: I'm leaning heavily on message propagation here.  The root logger won't be used directly, but its handler will catch messages from all the other loggers.

**Comment by mbarnes** [comment]
@ashcrow This is ready for review.

**Review by ashcrow — APPROVED** [review]

**Comment by ashcrow** [comment]
Manually squashed and merged.

**Pull request closed without merging**